OpenOversight/app/models/database.py

```diff
@@ -1085,35 +1085,35 @@ def is_active(self):
         """Override UserMixin.is_active to prevent disabled users from logging in."""
         return not self.disabled_at
 
-    def approve_user(self, approving_user_id: int):
+    def approve_user(self, approving_user_id: int, approved: bool):
         """Handle approving logic."""
-        if self.approved_at or self.approved_by:
-            return False
-
-        self.approved_at = datetime.now(timezone.utc)
-        self.approved_by = approving_user_id
-        db.session.add(self)
-        db.session.commit()
-        return True
-
-    def confirm_user(self, confirming_user_id: int):
+        if approved and not self.approved_by and not self.approved_at:
+            self.approved_at = datetime.now(timezone.utc)
+            self.approved_by = approving_user_id
+            db.session.commit()
```

Comment:
🔬 Are we likely to run into a scenario where we don't commit at any point?

Reply:
It's an expected case: if the user is already approved and the next form submission also has `approved` checked (or not approved and `approved` is not checked), then we don't want to update the existing user value